03-rnaseq/00-intro-to-rnaseq.Rmd

```diff
@@ -6,3 +6,125 @@ output:
+- [Computation for RNA-seq and ChIP-seq studies](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4121056/) [@Pepke2009].
+
+## About DESeq2
+
+DESeq2 is an R package that can normalize and handle RNA-seq data @Love2014.
+Our refine.bio data is summarized to the gene-level with tximport before you download it @Soneson2015. 
+In general, our examples suggest you download the data [non-quantile normalized](#about-quantile-normalization) so you can instead normalize the data with DESeq2, which requires you provide counts and *not* a normalized or corrected value like [TPMs](https://www.youtube.com/watch?v=TTUrtCY2k-w).
+
+### DESeq2 objects
+
+Many R Bioconductor packages have specialized object types they want your data to be formatted as. 
+For DESeq2, before we can use a lot the special functions, we need to get our data into a [`DESeqDataSet` object](https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/DESeqDataSet-class). 
+`DESeqDataSet` objects not only store your data, but additional transformations of your data, model information, etc. 
+
+From our refine.bio datasets, we will use a function `DESeqDataSetFromMatrix()` to create our [`DESeqDataSet` objects](https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/DESeqDataSet-class). 
+
+### DESeq2 normalization methods
+
+Although DESeq2 has multiple normalization methods, we generally stick to `vst()` (Variance Stablizing Transformation) or `rlog()`. 
+Both methods are very similar, and correct for [library size differences]() but `rlog()` sometimes takes a bit longer to run. 
+If you end up using a larger dataset and `rlog()` normalization takes a bit too long, you can switch to using `vst()` with confidence since previous data shows they yield similiar results given the dataset is large enough. 
+For more about DESeq2 normalization, we highly recommend this [StatsQuest video]((https://www.youtube.com/watch?v=UFB993xufUU)) which explains it quite nicely [@Starmer2017-deseq2].
+
+### Further resources for DESeq2
+
+- [StatsQuest: RPKM, FPKM and TPM, Clearly Explained!!!](https://www.youtube.com/watch?v=TTUrtCY2k-w) [@Starmer2015].
+- [StatsQuest: DESeq2, part 1, Library Normalization](https://www.youtube.com/watch?v=UFB993xufUU) [@Starmer2017-deseq2].
+- [DESeq2 vignette: Analyzing RNA-seq data with DESeq2](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html) [@Love2014].
+- [Beginner's guide to DESeq2](Bhttps://bioc.ism.ac.jp/packages/2.14/bioc/vignettes/DESeq2/inst/doc/beginner.pdf) [@Love2014-guide].
+- [Introduction to DGE - using DESeq2](https://hbctraining.github.io/DGE_workshop/lessons/04_DGE_DESeq2_analysis.html) [@dge-workshop].
+
+#### Why doesn't the gene I care about show up in this dataset?
+
+There can be a lot of reasons a particular gene might not show up in your refine.bio RNA-seq dataset, here are just a few reasons:  
```

Comment:
If they can't find the gene they want in their dataset is what I consider a gene "not showing up".

Reply:
Does that mean

> Not be present in the gene identifier column?

Reply:
We include transcripts that aren't detected at all?

Reply:
I don't think there's anything in our pipeline for what we're recommending here that would prevent those from being included (although it's been a bit since I've thought in depth about that). You can pull a handful RNA-seq experiments with QN skipped and see if you have rows with all zeroes. I expect the answer will be yes and isn't this why we include pre-filtering in some of our RNA-seq notebooks?

Reply:
Well, I could have seen a situation where none of the samples included had a non-zero value and that might be excluded from a refine.bio download, but perhaps if one sample expressed it it would still be maintained in the download. But, in the second instance, we'd still not want to rely on that for differential expression.

Reply:
But I just did a quick check and there are genes with a `rowSum()` of `0` are included so I'll delete these things.

Reply:
> Well, I could have seen a situation where none of the samples included had a non-zero value and that might be excluded from a refine.bio download, but perhaps if one sample expressed it it would still be maintained in the download. 

What does this mean?

Reply:
This was in response to: 
> isn't this why we include pre-filtering in some of our RNA-seq notebooks?

 I think the filtering we do would have still been useful/needed even if refine.bio only included a genes' data if at least one of the samples expressed it. But I didn't know refine.bio includes genes when no samples express it.

Reply:
Ah, gotcha!